targeteditor/src/main/scala/explore/targeteditor/MagnitudeForm.scala

```diff
@@ -3,44 +3,205 @@
-                  GridColumn(<.span(magnitude.value.toDoubleValue.toString)),
-                  GridColumn(<.span(magnitude.band.toString)),
-                  GridColumn(<.span(magnitude.system.toString))
-                )(^.paddingTop := "0.5rem", ^.paddingBottom := "0.5rem")
+      .getDerivedStateFromPropsAndState[State] { case (props, stateOpt) =>
+        val usedBands = HashSet.from(props.magnitudes.get.map(_.band))
+        stateOpt match {
+          case Some(state) if state.newBand.exists(b => !usedBands.contains(b)) =>
+            State.usedBands.set(usedBands)(state)
+          case _                                                                =>
+            State(usedBands, MagnitudeBand.all.diff(usedBands.toList).headOption)
+        }
+      }
+      .render { $ =>
+        val props = $.props
+        val state = $.state
+
+        AppCtx.withCtx { implicit ctx =>
+          val newBandView: Option[View[MagnitudeBand]] =
+            state.newBand.map(band =>
+              ViewF(band,
+                    (mod: MagnitudeBand => MagnitudeBand) =>
+                      $.modStateIn[IO](State.newBand.composePrism(some).modify(mod))
               )
             )
-          )(^.maxWidth := "250px")
-        )
+
+          Form(size = Small)(
+            <.label("Magnitudes"),
+            Segment(
+              Table(compact = TableCompact.Very)(
+                TableBody(
+                  props.magnitudes.get.toTagMod { magnitude =>
+                    def modInPlace[A](list: List[A])(find: A => Boolean)(mod: A => A): List[A] =
+                      list.indexWhere(find) match {
+                        case -1 => list
+                        case n  => (list.take(n) :+ mod(list(n))) ++ list.drop(n + 1)
+                      }
+
+                    // We are already focused in the magnitude we want
+                    val getMagnitude: List[Magnitude] => Magnitude = _ => magnitude
+
+                    def modMagnitude(mod: Magnitude => Magnitude)
+                      : List[Magnitude] => List[Magnitude] =
+                      list => modInPlace(list)(_.band === magnitude.band)(mod).sortBy(_.band)
+
+                    val magnitudeView: View[Magnitude] =
+                      props.magnitudes
+                        .zoom[Magnitude](getMagnitude)(modMagnitude)
+
+                    val magnitudeValueView: View[MagnitudeValue] =
+                      magnitudeView.zoom(Magnitude.value)
+
+                    val magnitudeBigDecimalView: View[BigDecimal] =
+                      magnitudeValueView.zoom(v => BigDecimal(v.toDoubleValue))(mod =>
+                        m => MagnitudeValue.fromDouble(mod(BigDecimal(m.toDoubleValue)).toDouble)
+                      )
+
+                    // TODO Add this to lucuma-ui?
+                    // val doubleValidFormat: ValidFormatInput[Double] = ???
+                    //
+                    // Actually we need something for an Int / 1000
```

Comment:
Ooops... This is an oversight. We need a good `ValidFormatInput` for `MagnitudeValue`s. I'll send this in another commit.